parser-typechecker/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1188,43 +1179,88 @@ displayRendered outputLoc pp =
   maybe displayOnly scratchAndDisplay outputLoc
   where
-  displayOnly = pure code
-  scratchAndDisplay path = do
-    path' <- canonicalizePath path
-    prependToFile code path'
-    pure (message code path')
-    where
-    prependToFile code path = do
-      existingContents <- do
-        exists <- doesFileExist path
-        if exists then readFile path
-        else pure ""
-      writeFile path . Text.pack . P.toPlain 80 $
-        P.lines [ code, ""
-                , "---- " <> "Anything below this line is ignored by Unison."
-                , "", P.text existingContents ]
-    message code path =
-      P.callout "☝️" $ P.lines [
-        P.wrap $ "I added these definitions to the top of " <> fromString path,
-        "",
-        P.indentN 2 code,
-        "",
-        P.wrap $
-          "You can edit them there, then do" <> makeExample' IP.update <>
-          "to replace the definitions currently in this namespace."
-      ]
-  code = displayDefinitions' ppe types terms
+    displayOnly = pure code
+    scratchAndDisplay path = do
+      path' <- canonicalizePath path
+      prependToFile code path'
+      pure (message code path')
+      where
+        prependToFile code path = do
+          existingContents <- do
+            exists <- doesFileExist path
+            if exists
+              then readFile path
+              else pure ""
+          writeFile path . Text.pack . P.toPlain 80 $
+            P.lines
+              [ code,
+                "",
+                "---- " <> "Anything below this line is ignored by Unison.",
+                "",
+                P.text existingContents
+              ]
+        message code path =
+          P.callout "☝️" $
+            P.lines
+              [ P.wrap $ "I added these definitions to the top of " <> fromString path,
+                "",
+                P.indentN 2 code,
+                "",
+                P.wrap $
+                  "You can edit them there, then do" <> makeExample' IP.update
+                    <> "to replace the definitions currently in this namespace."
+              ]
+    code =
+      P.syntaxToColor $ P.sep "\n\n" (prettyTypes <> prettyTerms)
+      where
+        ppeBody r = PPE.declarationPPE ppe r
+        ppeDecl = PPE.unsuffixifiedPPE ppe
+        prettyTerms =
+          map go . Map.toList $
+            -- sort by name
+            Map.mapKeys (first (PPE.termName ppeDecl . Referent.Ref) . dupe) terms
+        prettyTypes =
+          map go2 . Map.toList $
+            Map.mapKeys (first (PPE.typeName ppeDecl) . dupe) types
+        go ((n, r), dt) =
+          case dt of
+            MissingObject r -> missing n r
+            BuiltinObject typ ->
+              P.hang
+                ("builtin " <> prettyHashQualified n <> " :")
+                (TypePrinter.prettySyntax (ppeBody r) typ)
+            UserObject tm -> TermPrinter.prettyBinding (ppeBody r) n tm
+        go2 ((n, r), dt) =
```

Comment:
Once we hit `go2` I'd love a slightly more descriptive name and preferably also a type signature 😄

Reply:
Re: this comment and the following, this block was only formatted, but unchanged in this PR

Reply:
Maybe a bad merge? This code is definitely added according to the diff.

Reply:
Oh, sorry - I forgot that it did change, a little bit. We deleted an `Output` constructor that wasn't used, then inlined `displayDefinitions'`. But other than that, no changes.